ddl/db_test.go

```diff
@@ -5076,6 +5146,34 @@ func (s *testDBSuite1) TestModifyColumnTime(c *C) {
 		{"year", `70`, "timestamp", "", errno.ErrTruncatedWrongValue},
 		{"year", `99`, "timestamp", "", errno.ErrTruncatedWrongValue},
 	}
+	testModifyColumnTime(c, s.store, tests)
+}
+
+type testModifyColumnTimeCase struct {
+	from   string
+	value  string
+	to     string
+	expect string
+	err    uint16
+}
+
+func testModifyColumnTime(c *C, store kv.Storage, tests []testModifyColumnTimeCase) {
+	limit := variable.GetDDLErrorCountLimit()
+	variable.SetDDLErrorCountLimit(3)
+
+	tk := testkit.NewTestKit(c, store)
+	tk.MustExec("use test_db")
+	enableChangeColumnType := tk.Se.GetSessionVars().EnableChangeColumnType
+	tk.Se.GetSessionVars().EnableChangeColumnType = true
+
+	// Set time zone to UTC.
+	originalTz := tk.Se.GetSessionVars().TimeZone
+	tk.Se.GetSessionVars().TimeZone = time.UTC
+	defer func() {
+		variable.SetDDLErrorCountLimit(limit)
+		tk.Se.GetSessionVars().EnableChangeColumnType = enableChangeColumnType
+		tk.Se.GetSessionVars().TimeZone = originalTz
+	}()
 
 	for _, t := range tests {
 		tk.MustExec("drop table if exists t_mc")
```

Comment:
Tablename duplicated for them, plz fix.

Reply:
According the results of ``check_dev`` and ``check_dev_2 ``.
Seems tablename can be the same in different test case.

Reply:
Testcases could be paralleled, under which case it will fail... But I'll give you LGTM anyway.